File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int product=1;
        int i=0;
        vector<int> ans(0,nums.size());
        int containsZero=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==0){
                containsZero++;
            }else{
                 product=product*nums[i];
            }
        }
     
        if(containsZero>0){
            for(int i=0;i<nums.size();i++){
                if(nums[i]==0 && containsZero==1){
                    ans.push_back(product);
                }else{
                    ans.push_back(0);
                }
            }
        }else{
            for(int i=0;i<nums.size();i++){
                 ans.push_back(product/nums[i]);
            }
        }
        return ans;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] ans = new int[n];
        int product = 1;
        int isZeroPresent=0;
        for(int i=0;i<n;i++){
            if(nums[i]!=0){
                product*=nums[i];
            }else{
                isZeroPresent++;
            }
        }
        if(isZeroPresent>0){
            for(int i=0;i<n;i++){
                ans[i]=nums[i]==0 ? product :0;
            }
            return ans;
        }
        for(int i=0;i<n;i++){
            ans[i]=product/nums[i];
        }
        return ans;
    }
}